actions/pacstrap_action.go

```diff
@@ -0,0 +1,129 @@
+/*
+Pacstrap Action
+
+Construct the target rootfs with pacstrap tool.
+
+	# Yaml syntax:
+	- action: pacstrap
+	  config: <in-tree pacman.conf file>
+	  mirror: <in-tree mirrorlist file>
+
+Mandatory properties:
+
+  - config -- the pacman.conf file which will be used through the process
+  - mirror -- the mirrorlist file which will be used through the process
+*/
+package actions
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/go-debos/debos"
+	"github.com/go-debos/fakemachine"
+)
+
+type PacstrapAction struct {
+	debos.BaseAction `yaml:",inline"`
+	Config           string `yaml:"config"`
+	Mirror           string `yaml:"mirror"`
+}
+
+func (d *PacstrapAction) listOptionFiles(context *debos.DebosContext) ([]string, error) {
+	files := []string{}
+
+	if d.Config == "" {
+		return nil, fmt.Errorf("No config file set, aborting.")
+	}
+	d.Config = debos.CleanPathAt(d.Config, context.RecipeDir)
+	files = append(files, d.Config)
+
+	if d.Mirror == "" {
+		return nil, fmt.Errorf("No mirror file set, aborting.")
+	}
+	d.Mirror = debos.CleanPathAt(d.Mirror, context.RecipeDir)
+	files = append(files, d.Mirror)
+
+	return files, nil
+}
+
+func (d *PacstrapAction) Verify(context *debos.DebosContext) error {
+	files, err := d.listOptionFiles(context)
+	if err != nil {
+		return err
+	}
+
+	// Check if all needed files exists
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *PacstrapAction) PreMachine(context *debos.DebosContext, m *fakemachine.Machine, args *[]string) error {
+	mounts, err := d.listOptionFiles(context)
+	if err != nil {
+		return err
+	}
+
+	// Mount configuration files outside of recipes directory
+	for _, mount := range mounts {
+		m.AddVolume(path.Dir(mount))
+	}
+
+	return nil
+}
+
+func (d *PacstrapAction) Run(context *debos.DebosContext) error {
+	d.LogStart()
+
+	files := map[string]string{
+		"/etc/pacman.conf":         d.Config,
+		"/etc/pacman.d/mirrorlist": d.Mirror,
+	}
+
+	// Copy the config/mirrorlist files
+	for dest, src := range files {
+		if err := os.MkdirAll(path.Dir(dest), 0755); err != nil {
+			return err
+		}
+
+		read, err := ioutil.ReadFile(src)
+		if err != nil {
+			return err
+		}
+
+		if err = ioutil.WriteFile(dest, read, 0644); err != nil {
+			return err
+		}
+	}
+
+	// Setup the local keychain, within the fakemachine instance, since we
```

Comment:
what happens if this is run with `--disable-fakemachine` ? does it stamp on the host file ?

Reply:
No idea to be honest - `fakemachine` is the only option that has been tried.

Unless this is a blocker, I'm leaning that we can error out if run w/o `fakemachine`. Are you comfortable with that?

Reply:
> Unless this is a blocker, I'm leaning that we can error out if run w/o `fakemachine`. Are you comfortable with that?

You can do that as long as the CI tests pass ;-). We can add non-fakemachine support as a follow-up.

Reply:
At a glance the least invasive approach is to:
 - save the context type within `debos.DebosContext`
 - in the `Verify` hooks for `pacstrap/pacman` will print "Unsupported backend"
 - where the `Run` hooks will silently error out

It does feel a bit hacky, but the alternative is quite CI churn heavy. The current testing has no "known failing" concept. 

Does the above sound reasonable or do you have a particular approach in mind?

Reply:
I think it'd be great to first answer the question of what could happen to the user's machine if pacman/pacstrap action is run with no fakemachine

but your approach sounds ok; but instead of having to add a separate field to context and running in `Verify` you could just error out in `PreNoMachine` instead ;-)

Reply:
Yes, when run without fakemachine the current pacstrap stage will try and override the system files - pacman.conf and mirrorlist. Assuming debos has enough permissions (run as root or otherwise) it will succeed.

Added an error out in `PreNoMachine` although that is causing the existing CI to choke. Any pointers would be apprecaited.

Reply:
Fixed with the "include" snippet that you're shared. Thank you :bowing_man: